tests/test_in_out.py

```diff
@@ -253,3 +253,19 @@ def test_spinner_hiding_with_context_manager_and_exception():
     assert not sp._hide_spin.is_set()
 
     sp.stop()
+
+
+def test_write_non_str_objects(capsys):
+    objs = [
+        ("foo", "foo"),
+        (dict(cat="meow"), "{'cat': 'meow'}"),
+        (23, "23"),
+        (["foo", "bar", "'", 23], """['foo', 'bar', "'", 23]"""),
+    ]
+
+    sp = yaspin()
+    capsys.readouterr()
+    for obj, obj_str in objs:
```

Comment:
Let's avoid `for` loop here and run `objs` cases via `pytest.mark.parametrize`, this will run and report every case as a separate test. Check `tests/` for examples.

Reply:
Good idea, I'll look into it.